networks/oplib/resnet50.cc

```diff
@@ -0,0 +1,424 @@
+  std::vector<edsl::Tensor> B_block4c = {B[31], B[32], B[33]};
+  auto block_4c = block(block_4b, W_block4c, B_block4c, {1, 1}, false, "res4c");
+
+  std::vector<edsl::Tensor> W_block4d = {W[34], W[35], W[36]};
+  std::vector<edsl::Tensor> B_block4d = {B[34], B[35], B[36]};
+  auto block_4d = block(block_4c, W_block4d, B_block4d, {1, 1}, false, "res4d");
+
+  std::vector<edsl::Tensor> W_block4e = {W[37], W[38], W[39]};
+  std::vector<edsl::Tensor> B_block4e = {B[37], B[38], B[39]};
+  auto block_4e = block(block_4d, W_block4e, B_block4e, {1, 1}, false, "res4e");
+
+  std::vector<edsl::Tensor> W_block4f = {W[40], W[41], W[42]};
+  std::vector<edsl::Tensor> B_block4f = {B[40], B[41], B[42]};
+  auto block_4f = block(block_4e, W_block4f, B_block4f, {1, 1}, false, "res4f");
+
+  // 5
+  std::vector<edsl::Tensor> W_block5a = {W[43], W[44], W[45], W[46]};
+  std::vector<edsl::Tensor> B_block5a = {B[43], B[44], B[45], B[46]};
+  auto block_5a = block(block_4f, W_block5a, B_block5a, {2, 2}, true, "res5a");
+
+  std::vector<edsl::Tensor> W_block5b = {W[47], W[48], W[49]};
+  std::vector<edsl::Tensor> B_block5b = {B[47], B[48], B[49]};
+  auto block_5b = block(block_5a, W_block5b, B_block5b, {1, 1}, false, "res5b");
+
+  std::vector<edsl::Tensor> W_block5c = {W[50], W[51], W[52]};
+  std::vector<edsl::Tensor> B_block5c = {B[50], B[51], B[52]};
+  auto block_5c = block(block_5b, W_block5c, B_block5c, {1, 1}, false, "res5c");
+
+  // End
+  auto global_mean = op::mean(block_5c, edsl::make_tuple<int64_t>({1, 2}));
+  auto W_dense = W[53];
+  auto B_dense = B[53];
+  auto dense = op::dot(global_mean, W_dense) + B_dense;
+  auto softmax = op::softmax(dense, 1);
+  edsl::Program program("resnet50", {softmax});
+  return program;
+}
+
+void run_resnet50(int64_t batch_size = 1, size_t iterations = 100, size_t report_period = 10) {
+  auto I = edsl::Placeholder(PLAIDML_DATA_FLOAT32, {batch_size, 224, 224, 3});
+  auto W = weight_placeholders();
+  auto B = bias_placeholders();
+  IVLOG(1, "About to build resnet50 program");
+  auto program = resnet50_program(batch_size, I, W, B);
+  IVLOG(1, "Program built");
+  auto all_inputs = W;
+  all_inputs.insert(all_inputs.end(), B.begin(), B.end());
+  all_inputs.push_back(I);
+  auto bound_program = exec::Binder(program).set_inputs(all_inputs).build();
+  IVLOG(1, "Inputs bound");
+  bound_program->run();
+  IVLOG(1, "Initial run complete");
+  for (size_t i = 0; i < iterations; ++i) {
+    bound_program->run();
+    if (i % report_period == report_period - 1) {
+      IVLOG(1, "Run " << i + 1 << " complete");
+    }
+  }
+}
+
+}  // namespace plaidml::networks
+
+int main(int argc, char* argv[]) {
+  using plaidml::networks::run_resnet50;
+
+  try {
+    auto level_str = vertexai::env::Get("PLAIDML_VERBOSE");
+    if (level_str.size()) {
+      auto level = std::atoi(level_str.c_str());
+      if (level) {
+        el::Loggers::setVerboseLevel(level);
+      }
+    }
```

Comment:
This should be happening already once you call `plaidml::op::init()`, see: https://github.com/plaidml/plaidml/blob/master/plaidml2/core/ffi.cc#L54